code/game/objects/effects/anomalies/anomalies_ectoplasm.dm

```diff
@@ -0,0 +1,211 @@
+/obj/effect/anomaly/ectoplasm
+	name = "ectoplasm anomaly"
+	desc = "It looks like the souls of the damned are trying to break into the realm of the living again. How upsetting."
+	icon_state = "ectoplasm"
+	aSignal = /obj/item/assembly/signaler/anomaly/ectoplasm
+	lifespan = 100 SECONDS //This one takes slightly longer, because it can run away.
+
+	///Blocks the anomaly from updating ghost count. Used in case an admin wants to rig the anomaly to be a certain size or intensity.
+	var/override_ghosts = FALSE
+	///The numerical power of the anomaly. Calculated in anomalyEffect. Also used in determining the category of detonation effects.
+	var/effect_power = 0
+	///The actual number of ghosts orbiting the anomaly.
+	var/ghosts_orbiting = 0
+
+/obj/effect/anomaly/ectoplasm/Initialize(mapload, new_lifespan, drops_core)
+	. = ..()
+
+	AddComponent(/datum/component/deadchat_control/cardinal_movement, ANARCHY_MODE, list(), 7 SECONDS)
+
+	if(. == COMPONENT_INCOMPATIBLE)
+		return
+
+/obj/effect/anomaly/ectoplasm/examine(mob/user)
+	. = ..()
+
+	if(isobserver(user))
+		. += span_info("Orbiting this anomaly will increase the size and intensity of its effects.")
+
+/obj/effect/anomaly/ectoplasm/examine_more(mob/user)
+	. = ..()
+
+	switch(effect_power)
+		if(0 to 25)
+			. += span_notice("The space around the anomaly faintly resonates. It doesn't seem very powerful at the moment.")
+		if(26 to 49)
+			. += span_notice("The space around the anomaly seems to vibrate, letting out a noise that sounds like ghastly moaning. Someone should probably do something about that.")
+		if(50 to 100)
+			. += span_alert("The anomaly pulsates heavily, about to burst with unearthly energy. This can't be good.")
+
+/obj/effect/anomaly/ectoplasm/anomalyEffect(delta_time)
+	. = ..()
+
+	if(!override_ghosts)
```

Comment:
Do an early return here, like

```txt
if(override_ghosts)
    return

(rest of the code goes here)
```

better for the indentation later on and makes more logical sense with what you want ghost_override to be/do.